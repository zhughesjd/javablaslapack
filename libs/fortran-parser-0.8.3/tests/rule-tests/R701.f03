! Test primary, which is
!      primary  is  constant
!               or  designator
!               or  array-constructor
!               or  structure-constructor
!               or  function-reference
!               or  type-param-inquiry
!               or  type-param-name
!               or  ( expr )
!
! Each alternative is tested separately in the corresponding rule tests.
! primary is tested as an rhs.
a = 3
a = a
a = (/3,3/)
a = b(3,2,1)
a = func(1)
a = b%dim
a = (3)

end

