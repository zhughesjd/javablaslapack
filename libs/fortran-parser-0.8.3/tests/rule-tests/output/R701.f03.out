*******************************************
args[1]: R701.f03
R1101:main-program__begin:
R 204:specification-part: numUseStmts=0 numImportStmts=0 numDeclConstructs=0
R 613:part-ref: id=[@24,414:414='a',<209>,13:0] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 603:designator: hasSubstringRange=false
R 601:variable:
R 406:int-literal-constant: digitString=[@28,418:418='3',<10>,13:4] kindParam=null
R 306:literal-constant:
R 701:primary:
R 734:assignment-stmt:
R 214:action-stmt:
R 213:executable-construct:
R 613:part-ref: id=[@30,420:420='a',<209>,14:0] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 603:designator: hasSubstringRange=false
R 601:variable:
R 613:part-ref: id=[@34,424:424='a',<209>,14:4] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R -3:designator-or-func-ref:
R 701:primary:
R 734:assignment-stmt:
R 214:action-stmt:
R 213:executable-construct:
R 209:execution-part-construct:
R 613:part-ref: id=[@36,426:426='a',<209>,15:0] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 603:designator: hasSubstringRange=false
R 601:variable:
R 469:ac-value-list__begin:
R 406:int-literal-constant: digitString=[@42,432:432='3',<10>,15:6] kindParam=null
R 306:literal-constant:
R 701:primary:
R 469:ac-value:
R 406:int-literal-constant: digitString=[@44,434:434='3',<10>,15:8] kindParam=null
R 306:literal-constant:
R 701:primary:
R 469:ac-value:
R 469:ac-value-list: count=2
R 466:ac-spec:
R 465:array-constructor:
R 701:primary:
R 734:assignment-stmt:
R 214:action-stmt:
R 213:executable-construct:
R 209:execution-part-construct:
R 613:part-ref: id=[@48,438:438='a',<209>,16:0] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 603:designator: hasSubstringRange=false
R 601:variable:
R 619:section-subscript-list__begin:
R 406:int-literal-constant: digitString=[@54,444:444='3',<10>,16:6] kindParam=null
R 306:literal-constant:
R 701:primary:
R 619:section-subscript: hasLowerBound=true hasUpperBound=false hasStride=false isAmbiguous=true
R 406:int-literal-constant: digitString=[@56,446:446='2',<10>,16:8] kindParam=null
R 306:literal-constant:
R 701:primary:
R 619:section-subscript: hasLowerBound=true hasUpperBound=false hasStride=false isAmbiguous=true
R 406:int-literal-constant: digitString=[@58,448:448='1',<10>,16:10] kindParam=null
R 306:literal-constant:
R 701:primary:
R 619:section-subscript: hasLowerBound=true hasUpperBound=false hasStride=false isAmbiguous=true
R 619:section-subscript-list: count=3
R 613:part-ref: id=[@52,442:442='b',<209>,16:4] hasSelectionSubscriptList=true hasImageSelector=false
R 612:data-ref: numPartRef=1
R -3:designator-or-func-ref:
R 701:primary:
R 734:assignment-stmt:
R 214:action-stmt:
R 213:executable-construct:
R 209:execution-part-construct:
R 613:part-ref: id=[@61,451:451='a',<209>,17:0] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 603:designator: hasSubstringRange=false
R 601:variable:
R 619:section-subscript-list__begin:
R 406:int-literal-constant: digitString=[@67,460:460='1',<10>,17:9] kindParam=null
R 306:literal-constant:
R 701:primary:
R 619:section-subscript: hasLowerBound=true hasUpperBound=false hasStride=false isAmbiguous=true
R 619:section-subscript-list: count=1
R 613:part-ref: id=[@65,455:458='func',<209>,17:4] hasSelectionSubscriptList=true hasImageSelector=false
R 612:data-ref: numPartRef=1
R -3:designator-or-func-ref:
R 701:primary:
R 734:assignment-stmt:
R 214:action-stmt:
R 213:executable-construct:
R 209:execution-part-construct:
R 613:part-ref: id=[@70,463:463='a',<209>,18:0] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 603:designator: hasSubstringRange=false
R 601:variable:
R 613:part-ref: id=[@74,467:467='b',<209>,18:4] hasSelectionSubscriptList=false hasImageSelector=false
R 613:part-ref: id=[@76,469:471='dim',<209>,18:6] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=2
R -3:designator-or-func-ref:
R 701:primary:
R 734:assignment-stmt:
R 214:action-stmt:
R 213:executable-construct:
R 209:execution-part-construct:
R 613:part-ref: id=[@78,473:473='a',<209>,19:0] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 603:designator: hasSubstringRange=false
R 601:variable:
R 406:int-literal-constant: digitString=[@83,478:478='3',<10>,19:5] kindParam=null
R 306:literal-constant:
R 701:primary:
R 701:primary:
R 734:assignment-stmt:
R 214:action-stmt:
R 213:executable-construct:
R 209:execution-part-construct:
R 208:execution-part:
R1103:end-program-stmt: id=null
R1101:main-program: hasProgramStmt=false hasExecutionPart=true hasInternalSubprogramPart=false
Parser exiting normally
********************************************
